### ISSUE TITLE
Carthage error: no shared framework schemes for any of the platforms: iOS

### ISSUE THREAD

**Issue by tleyden** [couchbase/couchbase-lite-ios#1097]
Steps to repro:

* Checkout master branch of couchbase-lite-ios
* Install carthage (brew install?)
* Create /tmp/carthagetest/Cartfile with contents: https://gist.github.com/tleyden/b8d025b05e2ac8fc5dc5
* Run `carthage update --platform ios`

Result:

```
$ carthage update --platform ios
*** Fetching couchbase-lite-ios
*** Checking out couchbase-lite-ios at "v1.495"
*** xcodebuild output can be found in /var/folders/6_/n30fk4vj36dggx3tt7m7cnxw0000gp/T/carthage-xcodebuild.EjqOKa.log
*** Skipped building couchbase-lite-ios due to the error:
Dependency "couchbase-lite-ios" has no shared framework schemes for any of the platforms: iOS

If you believe this to be an error, please file an issue with the maintainers at https://github.com/couchbase/couchbase-lite-ios/issues/new
```

**Comment by snej** [comment]
> *** Checking out couchbase-lite-ios at "v1.495"

That is an _ancient_ tag from March 2014. Why did it check that out?

**Comment by tleyden** [comment]
I think it might be related to this:

https://github.com/Carthage/Carthage#tag-stable-releases

**Comment by tleyden** [comment]
I think the following might work:

* Delete all the v1.x tags in this repo
* Running carthage should pull 1.1, which may fail, but ..
* Tag the repo with 1.2.0
* Running carthage should pull 1.2.0 at that point

**Comment by tleyden** [comment]
> Delete all the v1.x tags in this repo

In defense of Carthage, it is a bit confusing to have a `1.1.0` tag that is later than a `v1.495` .. I'm sure this has confused and/or annoyed many a user as well.

**Comment by snej** [comment]
Well, as you know, the versioning got reset in the change from TouchDB to Couchbase Lite. But that's history and we don't need those old tags. I've deleted them now.

**Comment by snej** [comment]
Can you tell Carthage to pull branch `release/1.2.0`, instead of a tag?

**Comment by tleyden** [comment]
Yeah, that allowed it to go further, but now it's getting a different error:

```
$ cat Cartfile
github "couchbase/couchbase-lite-ios" "release/1.2.0"
$ carthage update --platform ios
*** Fetching couchbase-lite-ios
*** Checking out couchbase-lite-ios at "42f3826b420512eca23c9c40dc8a6d3470efe50e"
*** xcodebuild output can be found in /var/folders/6_/n30fk4vj36dggx3tt7m7cnxw0000gp/T/carthage-xcodebuild.EaXvIU.log
A shell task failed with exit code 66:
xcodebuild: error: Scheme CBForest-iOS-Carthage is not currently configured for the build action.
```

**Comment by snej** [comment]
The target that scheme uses got replaced (as part of making tvOS build.)
The fix is trivial, and only affects that scheme's .xcscheme file, but we need approval from @zgramana to get it into 1.2. I'll wait to push it until then.

**Comment by tleyden** [comment]
> I'll wait to push it until then.

Looks like it didn't get into 1.2 -- did it get pushed to master?

**Comment by snej** [comment]
Zack gives thumbs-up to pushing it to the release/1.2 branch, and to updating the tag.

**Comment by snej** [comment]
I've updated the `release/1.2` branch, and moved the `1.2` tag to point to it. So it should be fixed now.

**Comment by jamieathans** [comment]
As specified in the README, I added the following to my Cartfile:

`github "couchbase/couchbase-lite-ios" "release/1.2.0"`

but still get the error

```
*** Checking out couchbase-lite-ios at "7f33ceb50bf78af5d7491d2571764598dca17b67"
*** Skipped building couchbase-lite-ios due to the error:
Dependency "couchbase-lite-ios" has no shared framework schemes for any of the platforms: iOS
```

anyone else still getting this?

**Comment by geodesicer** [comment]
yes jamieathens,

got exactly the same problem. Core Plot works nicely with Carthage, but have tried different changes in the xcode project file for CouchbaseLite to no avail.

Why bother? Well, since I have to make 6 dynamic frameworks (4 our own, CouchbaseLite and Core Plot)) to support iOS, tvOS and Mac and at the same time handle > 5 apps using these frameworks, I am trying to stop hand editing CBL (and the binary versions don't work for several reasons) to create the CBL frameworks for all the platforms - and not to end up in Archive code sign problem, slicing the frameworks through lipo etc etc.

Core Plot works so far, but that framework might have other problems that I have just not seen yet.
Will report back if I manage to solve these problems, just started out so will go on for a bit more.

**Comment by geodesicer** [comment]
I managed to get past the previous errors by setting the scheme "CBL iOS framework" to shared in Xcode, had no luck (yet) getting Carthage to find the "Carthage"-scheme's defined in CouchbaseLite project file - I presume those are provided specifically to support Carthage.

New error is now: 
*** Building scheme "CBL iOS framework" in CouchbaseLite.xcodeproj
** BUILD FAILED **

The following build commands failed:
	Ld .../Library/Developer/Xcode/DerivedData/CouchbaseLite-dzimwfjavrgcuifmbkesypyxjqmm/Build/Intermediates/CouchbaseLite.build/Release-iphoneos/CBL\ iOS\ framework.build/Objects-normal/arm64/CouchbaseLite normal arm64
(1 failure)

ld: warning: ignoring undefined symbol .objc_class_name_CBLJSViewCompiler from -exported_symbols_list
ld: warning: ignoring undefined symbol .objc_class_name_CBLJSViewCompiler from -exported_symbols_list

clang: error: linker command failed with exit code 1 (use -v to see invocation)
A shell task failed with exit code 65:
** BUILD FAILED **

Next step seems to be understanding how Carthage actually selects the scheme's from the project file.

**Comment by geodesicer** [comment]
Meanwhile, I got it to build by doing the following:

> cd  <you>/Carthage/Checkouts/couchbase-lite-ios
> xcodebuild -scheme "Continuous iOS" -project CouchbaseLite.xcodeproj build

...
Touch /Users/<you>/Library/Developer/Xcode/DerivedData/CouchbaseLite-dzimwfjavrgcuifmbkesypyxjqmm/Build/Products/Debug-iOS-universal/CouchbaseLite.framework
    cd /Users/<you>/<your_project_path>/Carthage/Checkouts/couchbase-lite-ios
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/Users/<you>/.rvm/gems/ruby-2.1.2/bin:/Users/<you>/.rvm/gems/ruby-2.1.2@global/bin:/Users/<you>/.rvm/rubies/ruby-2.1.2/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/git/bin:/Users/<you>/.rvm/bin:/Users/<you>/bin"
    /usr/bin/touch -c /Users/<you>/Library/Developer/Xcode/DerivedData/CouchbaseLite-dzimwfjavrgcuifmbkesypyxjqmm/Build/Products/Debug-iOS-universal/CouchbaseLite.framework

** BUILD SUCCEEDED **

**Comment by snej** [comment]
This still seems to be a problem (see dup #1219). Reopening.

**Comment by snej** [comment]
@pasin, was that commit of yours up above (925431d) supposed to be added to the `release/1.2` branch too?

**Comment by pasin** [comment]
I made the fix only on 1.2.1 branch but not 1.2.0. I could add that to 1.2.0 as well but as we are going to release 1.2.1 soon, should I still do that?

Tagged @zgramana as well.

**Comment by pasin** [comment]
Need to verify if we need to do the same fix for 1.3 or not.

**Comment by pasin** [comment]
I couldn't verify if this works from 1.3.0 as I got blocked by https://github.com/Carthage/Carthage/issues/1370. I tried to downgrade to v.1.6.2 but I got another weird error:

```
$ carthage build
*** xcodebuild output can be found in /var/folders/8d/dtkmgqg51f77g_1qg1w0lz7w0000gn/T/carthage-xcodebuild.9Q5GBE.log
Failed to read file or folder at /Users/pasin/Git/couchbase/Tests/TestCart/Cartfile.resolved: Error Domain=NSCocoaErrorDomain Code=260 "The file “Cartfile.resolved” couldn’t be opened because there is no such file." UserInfo={NSFilePath=/Users/pasin/Git/couchbase/Tests/TestCart/Cartfile.resolved, NSUnderlyingError=0x7f9949c428c0 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
```

**Comment by snej** [comment]
Mac build fails due to missing SQLCipher library — Carthage tries to build the scheme "CBL Mac+SQLCipher", but fails because there is no file `vendor/SQLCipher/libs/osx/libSQLCipher.a` — we require that it be manually copied into that directory.

**Comment by snej** [comment]
iOS build works — I ran `carthage build --platform iOS` with no errors, with a Cartfile pointed at the current master branch.

**Comment by snej** [comment]
https://github.com/Carthage/Carthage/issues/1209 is covering this … I filed it back in March but forgot to link to it here.

**Comment by snej** [comment]
I tried removing the "CBL Mac+SQLCipher" shared scheme, but that didn't help, because the "CI MacOS" scheme also builds the problematic target, and we can't remove that scheme because Jenkins needs it. :-(

**Comment by snej** [comment]
I'll close this issue because the original reported problem is gone. Will file a new one regarding the Mac build not working.

**Comment by snej** [comment]
New issue is #1344.

**Comment by rovins** [comment]
Hi, I got a solution this is working fine for me. I hope it will work.
1. Go to manage scheme
2. Check shared.

**Comment by bhajian** [comment]
I still have this issue

**Comment by awilliams88** [comment]
@rovins solution worked for me. Was having trouble while installing mapbox-arkit-ios using carthage. Thanks Bro :-)

**Comment by mayankwadhwa111** [comment]
rovins where to find the options "Manage Scheme" and then shared

**Comment by rovins** [comment]
@ mayankwadhwa111 just select your application name in XCode which you are trying to build, will show you a list like-edit scheme, new scheme and manage schemes so please select manage schemes you will get a sub window from there you can check shared check box for you target anand enjoy
